dll/win32/shell32/wine/brsfolder.c

```diff
@@ -576,15 +603,25 @@ static HRESULT BrsFolder_Treeview_Changed( browse_info *info, NMTREEVIEWW *pnmtv
     LPTV_ITEMDATA lptvid = (LPTV_ITEMDATA) pnmtv->itemNew.lParam;
     WCHAR name[MAX_PATH];
 
+#ifdef __REACTOS__
+    BOOL bBtnState;
+#endif
     ILFree(info->pidlRet);
     info->pidlRet = ILClone(lptvid->lpifq);
 
     if (GetName(lptvid->lpsfParent, lptvid->lpi, SHGDN_NORMAL, name))
             SetWindowTextW( GetDlgItem(info->hWnd, IDC_BROWSE_FOR_FOLDER_FOLDER_TEXT), name );
-
+#ifndef __REACTOS__
     browsefolder_callback( info->lpBrowseInfo, info->hWnd, BFFM_SELCHANGED,
                            (LPARAM)info->pidlRet );
+
     BrsFolder_CheckValidSelection( info, lptvid );
+#else
+    bBtnState = browsefolder_callback( info->lpBrowseInfo, info->hWnd, BFFM_SELCHANGED,
+                           (LPARAM)info->pidlRet );
+
+    BrsFolder_CheckValidSelection( info, lptvid, bBtnState);
+#endif
```

Comment:
Any plans to send this to Wine in the future? Maybe @darkshadow44 / Fabian Maurer (on Mattermost) could help with that.